### PR TITLE
storage capacity conformance

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

The test was added in https://github.com/kubernetes/kubernetes/pull/108445 and needs to be promoted to conformance once it has been run successfully for two weeks.

#### Which issue(s) this PR fixes:

See https://github.com/kubernetes/kubernetes/pull/108445#pullrequestreview-916583227

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: test/e2e/storage/csistoragecapacity.go [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+	/*
+		Release: v1.24
+		Testname: CSIStorageCapacity API
+		Description:
+		The storage.k8s.io API group MUST exist in the /apis discovery document.
+		The storage.k8s.io/v1 API group/version MUST exist in the /apis/mode.k8s.io discovery document.
+		The csistoragecapacities resource MUST exist in the /apis/storage.k8s.io/v1 discovery document.
+		The csistoragecapacities resource must support create, get, list, watch, update, patch, delete, and deletecollection.
+	*/
+	framework.ConformanceIt(" should support CSIStorageCapacities API operations", func() {
```
Comment (hh) [inline test/e2e/storage/csistoragecapacity.go:47]: This line can be merged after https://github.com/kubernetes/kubernetes/pull/108445/files#diff-5e1072eb7b333bc3faf7068eb5ce93d29adaad658b5bcdd52a8238c18c5f8f92R48 merges, and the resulting CI jobs are non-flaky for a couple weeks.

**Comment by pohly** [comment]
@hh @Riaankl : this conformance PR is ready for merging.

It needs to go into 1.24 to complement the promotion of the CSIStorageCapacity API to GA.

**Comment by Riaankl** [comment]
/assign @smarterclayton @johnbelamaric 
Can you please `/lgtm` & `/approve` the PR to and add to `milestone v1.24` before test freeze to prevent it from incurring new tech debt.

**Pull request merged**